lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java

```diff
@@ -114,10 +131,16 @@ public static Map<String, String> getElementValues(Path file, Set<String> elemen
         }
         Map<String, String> returnValues = new HashMap<String, String> ();
 
-        XMLInputFactory factory = XMLInputFactory.newInstance();
+        XMLInputFactory factory = getXmlInputFactory();
+        readElementValues(file, elementNames, factory, returnValues);
+        return returnValues;
+    }
+
+    private static void readElementValues(Path file, Set<String> elementNames, XMLInputFactory factory, Map<String, String> returnValues) {
         XMLEventReader reader = null;
         try {
             reader = factory.createXMLEventReader(new FileInputStream(file.toFile()));
+
```

Comment:
The creation of the FileInputStream needs to be pulled out of this constructor so that we can reference it and close it in the finally statement. We had missed putting this in a try with resources originally.

Reply:
> The creation of the FileInputStream needs to be pulled out of this constructor so that we can reference it and close it in the finally statement. We had missed putting this in a try with resources originally.

Done